utils/instructions/NftVoter/castNftVote.ts

```diff
@@ -0,0 +1,234 @@
+import { getAssociatedTokenAddress } from '@blockworks-foundation/mango-v4'
+import { DasNftObject } from '@hooks/queries/digitalAssets'
+import { fetchNFTbyMint } from '@hooks/queries/nft'
+import { NftVoterClient } from '@utils/uiTypes/NftVoterClient'
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+import { chunks } from '@utils/helpers'
+import {
+  NftVoteRecord,
+  getNftActionTicketProgramAddress,
+  getNftVoteRecordProgramAddress,
+} from 'NftVotePlugin/accounts'
+import { PROGRAM_ID as ACCOUNT_COMPACTION_PROGRAM_ID } from '@solana/spl-account-compression'
+import { SYSTEM_PROGRAM_ID } from '@solana/spl-governance'
+import { NftVoter } from 'idls/nft_voter'
+import { NftVoterV2 } from 'idls/nft_voter_v2'
+import { Program } from '@project-serum/anchor'
+import {
+  AccountData,
+  UpdateVoterWeightRecordTypes,
+} from '@utils/uiTypes/VotePlugin'
+import { getCnftParamAndProof } from 'NftVotePlugin/getCnftParamAndProof'
+
+export const getCastNftVoteInstruction = async (
+  client: NftVoterClient,
```

Comment:
we only ever use the `program` from the `client`, so that should be the parameter, not the client. I'm especially saying this because one day the vote client class / concept needs to be refactored.

now, the `program` object may not even be stateful, so it's unclear that even needs to be a parameter itself. but whatever, that could change in other versions of anchor maybe.